File: Day53/DSA/Maximum_Product_of_Two_Elements_in_an_Array.java
Question:-  https://leetcode.com/problems/maximum-product-of-two-elements-in-an-array/description/

Code:-  

class Solution {
    public int maxProduct(int[] nums) {
        int max = -1;

        int smax = -1;

        for(int i = 0;i<nums.length;i++)
        {
            if(nums[i] > max)
            {
                smax = max;
                max = nums[i];
            }
            else if(nums[i] > smax)
            {
                smax = nums[i];
            }
        }

        return (max-1) * (smax - 1);
    }
}

File: Day53/DSA/Maximum_Product_of_Two_Elements_in_an_Array.cpp
Question:-  https://leetcode.com/problems/maximum-product-of-two-elements-in-an-array/description/


Code:-

class Solution {
public:
    int maxProduct(vector<int>& nums) {

        priority_queue<int> pq;

        for (int i : nums) {
            pq.push(i);
        }

        int x = pq.top();
        pq.pop();
        int y = pq.top();

        return (x - 1) * (y - 1);
    }
};